crypto/threads_pthread.c

```diff
@@ -169,6 +169,25 @@ int CRYPTO_atomic_add(int *val, int amount, int *ret, CRYPTO_RWLOCK *lock)
     return 1;
 }
 
+int CRYPTO_atomic_read(int *val, int *ret, CRYPTO_RWLOCK *lock)
+{
+# if defined(__GNUC__) && defined(__ATOMIC_ACQ_REL)
+    if (__atomic_is_lock_free(sizeof(*val), val)) {
+        __atomic_load(val, ret, __ATOMIC_ACQUIRE);
```

Comment:
Is acquire the order you want to use here? Maybe acquire release?

Reply:
Acquire/release isn't legal for an atomic load.  The valid memory orders are: relaxed, seq_cst, acquire, consume.  Because this is reading but not writing, I think acquire is the correct ordering.